src/material/core/ripple/ripple.ts

```diff
@@ -123,7 +123,8 @@ export class MatRipple implements OnInit, OnDestroy, RippleTarget {
               @Optional() @Inject(MAT_RIPPLE_GLOBAL_OPTIONS) globalOptions?: RippleGlobalOptions,
               @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode?: string) {
 
-    this._globalOptions = globalOptions || {};
+    // Clone the global options so we don't mutate them by accident.
+    this._globalOptions = globalOptions ? {...globalOptions} : {};
```

Comment:
I think this breaks the case where people pass in a class instance for global options (which is something we also have documented). Should we just handle the animation mode below in the `rippleConfig` getter?